encodings/dict/src/ops.rs

```diff
@@ -6,8 +8,14 @@ use crate::DictArray;
 
 impl ArrayOperationsImpl for DictArray {
     fn _slice(&self, start: usize, stop: usize) -> VortexResult<ArrayRef> {
-        DictArray::try_new(self.codes().slice(start, stop)?, self.values().clone())
-            .map(|a| a.into_array())
+        let sliced_code = self.codes().slice(start, stop)?;
+        if sliced_code.is_encoding(ConstantEncoding.id()) {
```

Comment:
Why are we checking for ConstantEncoding instead of is_constant? The cost? I don't think the act of slicing would ever check is_constant, so I can't see why you'd get one back.